snappi_ixnetwork/trafficitem.py

```
@@ -743,5 +827,48 @@ def results(self, request):
                     except Exception:
                         # TODO print a warning maybe ?
                         pass
-
+                if name in flows_has_latency:
+                    self._construct_latency(flow_row, latency_mode, row)
+                if name in flows_has_timestamp:
+                    self._construct_timestamp(flow_row, row)
+                if name not in flows_has_loss:
+                    flow_row.pop('loss')
         return list(flow_rows.values())
+
+    def _construct_latency(self, flow_row, latency_mode, row):
+        if latency_mode == 'store_forward':
+            latency_map = TrafficItem._RESULT_LATENCY_STORE_FORWARD
+        else:
+            latency_map = TrafficItem._RESULT_LATENCY_CUT_THROUGH
+        latency_result = {}
+        for external_name, internal_name, external_type in latency_map:
+            if internal_name not in row.Columns:
+                continue
+            try:
+                self._set_result_value(latency_result, external_name,
+                                       row[internal_name], external_type)
+            except Exception:
+                self._api.warning("Issue at the time processing latency result")
+                pass
+        if len(latency_result) > 0:
+            flow_row['latency'] = latency_result
+    
+    def _construct_timestamp(self, flow_row, row):
+        timestamp_result = {}
+        for external_name, internal_name, external_type in TrafficItem._RESULT_TIMESTAMP:
+            if internal_name not in row.Columns:
+                continue
```

should raise an error here as well